src/OrchardCore.Modules/OrchardCore.Swagger/Startup.cs

```diff
@@ -16,10 +17,17 @@ public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IS
             app.UseSwagger();
             app.UseSwaggerUI(c =>
             {
+                var shellSettings = serviceProvider.GetRequiredService<ShellSettings>();
+                var routeTenantPrefix = string.Empty;
+                if(shellSettings.RequestUrlPrefix != null)
+                {
+                    routeTenantPrefix = $"{shellSettings.RequestUrlPrefix}/";
+                }
```

Comment:
I think here it is better to use the http context `PathBase` that contains the tenant prefix but also an eventual virtual folder. Because a tenant can bluid another tenant, in some case `PathBase` may contain the prefix of another tenant, e.g in the `ConfigureServices` method (see below what can be done instead).

But because a tenant can't build the pipeline of another one (at least right now), here that's ok to use `PathBase` in the `Configure()` method.

---

Just for infos, see in the media startup `ConfigureServices` how we use

    HttpContext?.Features.Get<ShellContextFeature>()?.OriginalPathBase ?? null

Here, `OriginalPathBase` doesn't contain the tenant prefix but an eventual virtual folder

Reply:
Okay cool.

For infos what i saw when using `PathString`.

- If it has a value it always starts with a `/`.
- If it has no value, its `ToString()` returns an empty string.
- You can safely add 2 `PathString`, even if the 1st one ends with a `/` or not, or if it has no value.

- One limitation i saw is when using `path1.StartsWithSegments(path2 ...)`, you have to take care that path2 doesn't end with a `/`.

So here i would just have done something like this without checking `HasValue`.

    var swaggerPath= httpContextAccessor.HttpContext.Request.PathBase + new PathString("/swagger");
    c.SwaggerEndpoint($"{swaggerPath}/{definition.Name}/swagger.json", definition.Name);

I would say that we don't need the other null checks (that i didn't use in the above examlpe) but not sure in which context this delgate can be used.

Reply:
Done.

Reply:
👍